openmc/region.py

```diff
@@ -281,6 +305,26 @@ class Union(Region):
     def __init__(self, *nodes):
         self.nodes = list(nodes)
 
+    def __iter__(self):
+        for n in self.nodes:
+            yield n
+
+    def __contains__(self, point):
+        """Check whether a point is contained in the region.
+
+        Parameters
+        ----------
+        point : 3-tuple of float
+            Cartesian coordinates, :math:`(x',y',z')`, of the point
+
+        Returns
+        -------
+        bool
+            Whether the point is in the region
+
+        """
+        return any(point in n for n in self.nodes)
```

Comment:
The use of `__contains__` everywhere makes this super expressive, I like it a lot!